server/src/main/java/io/druid/client/DirectDruidClient.java

```diff
@@ -590,6 +591,27 @@ private void init()
     {
       if (jp == null) {
         try {
+          // Safety for if we are in a FJP
+          ForkJoinPool.managedBlock(new ForkJoinPool.ManagedBlocker()
```

Comment:
I wonder how this affects to the query performance. It looks like directly calling `future.get()` is more efficient than wrapping it with this Blocker. Is there any available document about this?

Reply:
`java.util.concurrent.ForkJoinPool.ManagedBlocker` and `java.util.concurrent.ForkJoinPool` have information in their java docs. Can you explain more about what kind of information you are looking for?

Reply:
@jihoonson performance penalty if running outside of a ForkJoinPool should be minimal.

@drcrallen please extract an utility method